java_src/foam/dao/history/HistoryDAO.java

```diff
@@ -70,9 +69,8 @@ public FObject put(FObject obj) {
     FObject current = this.find(obj);
 
     // add new history record
-    String objectId = (String) ((PropertyInfo) obj.getClassInfo().getAxiomByName("objectId")).f(obj);
     HistoryRecord historyRecord = new HistoryRecord();
-    historyRecord.setObjectId(objectId);
+    historyRecord.setObjectId(obj.getClassInfo().getId());
```

Comment:
That's assuming that the object has a property called id. If it doesn't then a NPE will be thrown.